File: test-suite/src/test/java/io/micronaut/docs/server/routes/MyRoutes.java
package io.micronaut.docs.server.routes;
// tag::imports[]
import io.micronaut.context.ExecutionHandleLocator;
import io.micronaut.web.router.DefaultRouteBuilder;
import javax.inject.Inject;
import javax.inject.Singleton;
// end::imports[]

// tag::class[]
@Singleton
public class MyRoutes extends DefaultRouteBuilder { // <1>

    public MyRoutes(ExecutionHandleLocator executionHandleLocator, UriNamingStrategy uriNamingStrategy) {
        super(executionHandleLocator, uriNamingStrategy);
    }

    @Inject
    void issuesRoutes(IssuesController issuesController) { // <2>
        GET("/issues/show/{number}", issuesController, "issue", Integer.class); // <3>
    }
}
// end::class[]

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/server/routes/MyRoutes.kt
package io.micronaut.docs.server.routes

// tag::imports[]
import io.micronaut.context.ExecutionHandleLocator
import io.micronaut.web.router.DefaultRouteBuilder
import io.micronaut.web.router.RouteBuilder

import javax.inject.Inject
import javax.inject.Singleton
// end::imports[]

// tag::class[]
@Singleton
class MyRoutes(executionHandleLocator: ExecutionHandleLocator, uriNamingStrategy: RouteBuilder.UriNamingStrategy) :
        DefaultRouteBuilder(executionHandleLocator, uriNamingStrategy) { // <1>

    @Inject
    fun issuesRoutes(issuesController: IssuesController) { // <2>
        GET("/issues/show/{number}", issuesController, "issue", Int::class.java) // <3>
    }
}
// end::class[]
